meson.build

```diff
@@ -0,0 +1,8 @@
+project('c++ starter project', 'cpp',
+  default_options : ['warning_level=3',
```

Comment:
How does `warning_level` relate back to `-Wall` `-Wextra` etc, on GCC et al?

Reply:
Roughly:

    1: -Wall
    2: -Wall -Wextra
    3: -Wall -Wextra -Wpedantic

In addition we also add `-Winvalid-pch`.

We don't have a 4, but there has been discussion on adding one and it would correspond to `-Weverything` on Clang but GCC does not seem to have a preset of this type.

Reply:
Some way to set `/W4` on Windows would be nice via a built-in, since that is what I like to recommend for MSVC.